CMakeLists.txt

```diff
@@ -1309,22 +1321,36 @@ ELSE()
         SET(IMPL_TARGET soft_oal)
     ENDIF()
 
-    ADD_LIBRARY(${IMPL_TARGET} SHARED ${OPENAL_OBJS} ${ALC_OBJS})
+    ADD_LIBRARY(${IMPL_TARGET} SHARED)
     IF(WIN32)
         SET_TARGET_PROPERTIES(${IMPL_TARGET} PROPERTIES PREFIX "")
     ENDIF()
 ENDIF()
+
+TARGET_SOURCES(${IMPL_TARGET} PRIVATE ${OPENAL_OBJS} ${ALC_OBJS})
+
+TARGET_LINK_LIBRARIES(${IMPL_TARGET}
+    PRIVATE common ${LINKER_FLAGS} ${EXTRA_LIBS} ${MATH_LIB})
```

Comment:
Unfortunately you can't link to `common` like this when the openal implementation is static. A static library can't link to another library. At best, CMake will track a static library target's dependencies and automatically include them as needed when linking, but the generated static `libopenal.a` library should be stand-alone and not need a separate `libcommon.a` with it.

Reply:
Hmm, Sorry!
In this case, would it make sense just to abandon the target "common" and simply include the sources to the libraries?

Reply:
Normally I'd prefer not to, but since it's only actually needed for these two libraries now, I suppose I can't see the harm.